contracts/PartyBid.sol

```diff
@@ -462,83 +564,40 @@ contract PartyBid is ReentrancyGuardUpgradeable, ERC721HolderUpgradeable {
+        // equal to 2x the winning bid
+        uint256 _listPrice = RESALE_MULTIPLIER * highestBid;
+        // users receive tokens at a rate of 1:TOKEN_SCALE for each ETH they contributed that was ultimately spent
+        // partyDAO receives a percentage of the total token supply equivalent to TOKEN_FEE_BASIS_POINTS
+        // partyHost receives a percentage of the total token supply equivalent to partyHostBasisPoints
+        (uint256 _tokenSupply, uint256 _tokenFee, uint256 _partyHostAmount) = _getTokenInflationAmounts(totalSpent);
         // deploy fractionalized NFT vault
         uint256 vaultNumber =
             IERC721VaultFactory(tokenVaultFactory).mint(
                 name,
                 symbol,
                 nftContract,
                 tokenId,
-                valueToTokens(_totalSpent),
-                _totalSpent,
+                _tokenSupply,
+                _listPrice,
                 0
             );
         // store token vault address to storage
         tokenVault = IERC721VaultFactory(tokenVaultFactory).vaults(vaultNumber);
-        // transfer curator to null address
+        // transfer curator to null address (burn the curator role)
         ITokenVault(tokenVault).updateCurator(address(0));
-    }
-
-    // ============ Internal: Claim ============
-
-    /**
-     * @notice Calculate the amount of fractional NFT tokens owed to the contributor
-     * based on how much ETH they contributed towards the auction,
-     * and the amount of excess ETH owed to the contributor
-     * based on how much ETH they contributed *not* used towards the auction
-     * @param _contributor the address of the contributor
-     * @return _tokenAmount the amount of fractional NFT tokens owed to the contributor
-     * @return _ethAmount the amount of excess ETH owed to the contributor
-     */
-    function _calculateTokensAndETHOwed(address _contributor)
-        internal
-        view
-        returns (uint256 _tokenAmount, uint256 _ethAmount)
-    {
-        uint256 _totalContributed = totalContributed[_contributor];
-        if (partyStatus == PartyStatus.AUCTION_WON) {
-            // calculate the amount of this contributor's ETH
-            // that was used for the winning bid
-            uint256 _totalUsedForBid = _totalEthUsedForBid(_contributor);
-            if (_totalUsedForBid > 0) {
-                _tokenAmount = valueToTokens(_totalUsedForBid);
-            }
-            // the rest of the contributor's ETH should be returned
-            _ethAmount = _totalContributed - _totalUsedForBid;
-        } else {
-            // if the auction was lost, no ETH was spent;
-            // all of the contributor's ETH should be returned
-            _ethAmount = _totalContributed;
+        // transfer tokens to PartyDAO multisig
+        _transferTokens(partyDAOMultisig, _tokenFee);
+        // transfer tokens to token recipient
+        if (partyHost != address(0)) {
```

Comment:
should we also check that `_partyHostAmount` != `0`?

Reply:
it's checked inside the `_transferTokens` function